### PR TITLE
[Merged by Bors] - Ignore clippy 1.58

### PR DESCRIPTION
**Pull request by mockersf**
- Work around #3666 until a proper fix is done
- Also update duplicate dependencies list

### CODE REVIEW

**Comment by alice-i-cecile** [comment]
Reported upstream: https://github.com/rust-lang/rust-clippy/issues/8275

File: crates/bevy_ecs/src/world/mod.rs [hunk 1]
```diff
@@ -1159,6 +1159,7 @@ impl fmt::Debug for World {
     }
 }
 
+#[allow(clippy::non_send_fields_in_send_ty)]
```
Comment (alice-i-cecile) [inline crates/bevy_ecs/src/world/mod.rs:1162]: We should really have a comment here that this is in fact unsound, and linking to #3519.

**Comment by mockersf** [comment]
this is not to have a fix, but to ignore the lint and not block CI until a proper fix is done. ignoring the lint won't be more unsound that before 1.58, so it's a not-bad temp "fix"

File: crates/bevy_ecs/src/world/mod.rs (hunk 1 above)
Comment (mockersf) [reply crates/bevy_ecs/src/world/mod.rs:1163]: added links to https://github.com/bevyengine/bevy/issues/3666

**Review by alice-i-cecile — APPROVED** [review]

**Review by IceSentry — APPROVED** [review]

**Review by DJMcNab — APPROVED** [review]

**Pull request closed without merging**